tests/e2e/support/commands.js

```diff
@@ -34,14 +47,74 @@ Cypress.Commands.add('loadMatchFixtures', (matches) => {
   });
 });
 Cypress.Commands.add('requestGameList', () => {
-  return new Promise((resolve) => {
+  return new Cypress.Promise((resolve) => {
     io.socket.get('/game/getList', function () {
       return resolve();
     });
   });
 });
+/**
+ * Signs up two players, navigates home, creates game, subscribes, ready's up
+ * @param {boolean} alreadyAuthenticated: skips setup steps: db wipe, signup, navigate /
+ */
+Cypress.Commands.add('setupGameAsP0', (alreadyAuthenticated = false, isRanked = false) => {
+  cy.then(() => {
+    if (alreadyAuthenticated) {
+      return;
+    }
+    cy.wipeDatabase().then(() => {
+      cy.visit('/');
+      cy.signupPlayer(username, validPassword);
+    });
```

Comment:
I'm pretty sure this is the key to those intermittent failures we were seeing. In order to truly wait until the database is wiped, we need to use `.then()` and chain off it as a parent command. I can't say exactly why but it's certainly working.